manifests/server/config.pp

```diff
@@ -117,7 +117,7 @@
     }
 
   }
-  elsif ! $puppet::server_dynamic_environments {
+  elsif ! $puppet::server_dynamic_environments and !$puppet::server_directory_environments {
```

Comment:
Also in the other PR: https://github.com/theforeman/puppet-puppet/pull/286 discusses this change. Unless there are some arguments in favour I don't think we want this change.